### PR TITLE
ci: update buf config

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

Fix https://dev.azure.com/cncf/envoy/_build/results?buildId=111877&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39&l=107

> WARN	Repository "buf.build/beta/prometheus" is deprecated: This module has deprecated in favor of buf.build/prometheus/client-model. Please use that instead.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by daixiang0** [comment]
Also run at local:
```
INFO: Running command line: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/com_github_bufbuild_buf/bin/buf lint
Failure: the server hosted at that remote is unavailable.
/source
##[error]: TESTS FAILED:
##[error]:  main@ /source/ci/format_pre.sh :70 (buf)
```

Not sure it is network issue.

@phlax could you help confirm it?

File: api/buf.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: v1
 deps:
     - buf.build/googleapis/googleapis:d1a849b8f8304950832335723096e954
     - buf.build/opencensus/opencensus
-    - buf.build/beta/prometheus
+    - buf.build/prometheus/client-model
```
Comment (phlax) [inline api/buf.yaml:5]: i think this wont work - the proto itself needs to be updated

Comment (daixiang0) [reply api/buf.yaml:5]: Yeah, I am updating it at local, but seems I cannot access https://api.buf.build/.

Comment (phlax) [reply api/buf.yaml:5]: wierd - any ideas why ?

File: ci/format_pre.sh [hunk 2]
```diff
@@ -66,6 +66,7 @@ CURRENT=check_format
 "${ENVOY_SRCDIR}"/tools/code_format/check_format.py check || fix_format
 
 CURRENT=buf
+bazel run "${BAZEL_BUILD_OPTIONS[@]}" @com_github_bufbuild_buf//:bin/buf mod update --debug
```
Comment (phlax) [inline ci/format_pre.sh:69]: i guess this is debugging - but this should not happen here i think

File: api/buf.yaml (hunk 1 above)
Comment (daixiang0) [reply api/buf.yaml:5]: I am behind corporate proxy, let me establish it first.

Comment (phlax) [reply api/buf.yaml:5]: wierd that this didnt fail in CI - i will test again - i wasnt able to updated it when i tried before

**Comment by daixiang0** [comment]
@phlax I can not access that site, feel free to update dependence at your side then post PR.

**Pull request closed without merging**